### PR TITLE
Refactor for better code readability

### PR DESCRIPTION
**Pull request by warpech**
I have made several improvements that were discussed with @tomalec over Slack. These changes are intended to increase code readability, which is needed before we can do further work.

### CODE REVIEW

**Comment by warpech** [comment]
### Compatibility

I have searched Palindrom and Starcounter GitHub orgs, and could not find any usage of the APIs that I've changed.

### Proposed more changes

We could change all properties to begin with `_`, because they are intended for private use. Private methods do begin with `_`.

### Benchmarks

#### Before this PR

```
jsonpatcherproxy generate operation
 Ops/sec: 136838 ±2.76% Ran 8652 times in 0.063 seconds.
jsonpatch generate operation
 Ops/sec: 109856 ±6.94% Ran 6494 times in 0.059 seconds.
jsonpatcherproxy mutation - huge object
 Ops/sec: 1283 ±0.26% Ran 67 times in 0.052 seconds.
jsonpatch mutation - huge object
 Ops/sec: 887 ±1.73% Ran 46 times in 0.052 seconds.
jsonpatcherproxy generate operation - huge object
 Ops/sec: 1084 ±0.99% Ran 58 times in 0.054 seconds.
jsonpatch generate operation - huge object
 Ops/sec: 860 ±1.65% Ran 47 times in 0.055 seconds.
PROXIFY big object
 Ops/sec: 2536 ±0.40% Ran 131 times in 0.052 seconds.
DIRTY-OBSERVE big object
 Ops/sec: 1746 ±1.18% Ran 94 times in 0.054 seconds.
```

#### After this PR

```
jsonpatcherproxy generate operation
 Ops/sec: 145558 ±2.37% Ran 9371 times in 0.064 seconds.
jsonpatch generate operation
 Ops/sec: 107443 ±5.93% Ran 6450 times in 0.060 seconds.
jsonpatcherproxy mutation - huge object
 Ops/sec: 923 ±1.95% Ran 52 times in 0.056 seconds.
jsonpatch mutation - huge object
 Ops/sec: 878 ±1.48% Ran 46 times in 0.052 seconds.
jsonpatcherproxy generate operation - huge object
 Ops/sec: 826 ±1.97% Ran 48 times in 0.058 seconds.
jsonpatch generate operation - huge object
 Ops/sec: 874 ±1.65% Ran 48 times in 0.055 seconds.
PROXIFY big object
 Ops/sec: 1845 ±2.16% Ran 103 times in 0.056 seconds.
DIRTY-OBSERVE big object
 Ops/sec: 1738 ±1.10% Ran 94 times in 0.054 seconds.
```

**Comment by warpech** [comment]
To verify that the TS type definitions are correct, I created a file locally `src/test.ts` with the following content and checked that it compiles:

```ts
import JSONPatcherProxy from '../';

var obj = {};
var instance = new JSONPatcherProxy(obj);
instance.generate();
instance.observe(true);
```

File: src/jsonpatcherproxy.js [hunk 1]
```diff
@@ -375,16 +337,33 @@ const JSONPatcherProxy = (function() {
    * Revokes all proxies rendering the observed object useless and good for garbage collection @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/revocable}
```
Comment (alshakero) [inline src/jsonpatcherproxy.js]: The comment could use a `,` after `Revokes all proxies`

File: src/jsonpatcherproxy.js [hunk 2]
```diff
@@ -216,73 +195,58 @@ const JSONPatcherProxy = (function() {
...
-    const revocableInstance = Proxy.revocable(obj, traps);
-    // cache traps object to disable them later.
-    revocableInstance.trapsInstance = traps;
-    revocableInstance.originalObject = obj;
+    const treeMetadata = Proxy.revocable(tree, handler);
+    // cache the object that contains traps to disable them later.
+    treeMetadata.handler = handler;
+    treeMetadata.originalObject = tree;
 
     /* keeping track of object's parent and path */
```
Comment (alshakero) [inline src/jsonpatcherproxy.js]: The `path` in the comment is outdated, should be `key`.

File: src/jsonpatcherproxy.js [hunk 3]
```diff
@@ -177,35 +168,23 @@ const JSONPatcherProxy = (function() {
             it is a good idea to remove this flag if we come across it here, in deleteProperty trap.
```
Comment (alshakero) [inline src/jsonpatcherproxy.js]: The `deleteProperty` in the comment is outdated now. Should be `trapForDeleteProperty`.

File: src/jsonpatcherproxy.js [hunk 4]
```diff
@@ -91,84 +88,78 @@ const JSONPatcherProxy = (function() {
...
       }
       operation.value = newValue;
     }
-    const reflectionResult = Reflect.set(target, key, newValue);
+    const reflectionResult = Reflect.set(tree, key, newValue);
     instance.defaultCallback(operation);
     return reflectionResult;
   }
   /**
    * A callback to be used as th proxy delete trap callback.
```
Comment (alshakero) [inline src/jsonpatcherproxy.js]: `the`*

File: src/jsonpatcherproxy.js [hunk 5]
```diff
@@ -91,84 +88,78 @@ const JSONPatcherProxy = (function() {
...
    * It updates parenthood map if needed, calls default callbacks with the changes occurred.
```
Comment (alshakero) [inline src/jsonpatcherproxy.js]: calls default callback*

**Comment by warpech** [comment]
Thank you for the reviews! Merging after the tests pass, as all the comments have been addressed

**Comment by alshakero** [comment]
It seems `.vscode/settings.json` was added unintentionally.

**Comment by warpech** [comment]
I actually did that intentionally, thinking that a common project config might be useful. Do you think it is better to remove it? @tomalec?

**Comment by tomalec** [comment]
I'm not sure. Usually, I'm not a fan of bloating project repo with IDE configs, but If think it's better to keep it, I'm good.

**Pull request merged**